src/move/rewards/sources/emojicoin_dot_fun_rewards.move

```diff
@@ -0,0 +1,102 @@
+module rewards::emojicoin_dot_fun_rewards {
+
+    use std::signer;
+    use aptos_framework::account::{Self, SignerCapability};
+    use aptos_framework::aptos_account;
+    use aptos_framework::aptos_coin::AptosCoin;
+    use aptos_framework::coin;
+    use aptos_framework::event;
+    use aptos_framework::randomness;
+    use emojicoin_dot_fun::emojicoin_dot_fun::{Self, Swap};
+
+    /// Resource account address seed for the vault.
+    const VAULT: vector<u8> = b"Vault";
+
+    /// Flat integrator fee.
+    const INTEGRATOR_FEE_RATE_BPS: u8 = 100;
+
+    // APT amounts.
+    const REWARD_AMOUNT_IN_OCTAS: u64 = 100_000_000;
+    const VOLUME_THRESHOLD_IN_OCTAS: u64 = 990_000_000;
+
+    // Randomness amounts.
+    const WIN_PERCENTAGE_NUMERATOR: u64 = 980;
+    const WIN_PERCENTAGE_DENOMINATOR: u64 = 10_000;
+
+    struct RewardsVaultSignerCapability has key {
+        signer_capability: SignerCapability,
+    }
+
+    #[event]
+    struct EmojicoinDotFunRewardsLotteryWinner has copy, drop, store {
+        swap: Swap,
+        reward_amount: u64,
+    }
+
+    #[randomness]
+    entry fun swap_with_rewards<Emojicoin, EmojicoinLP>(
+        swapper: &signer,
+        market_address: address,
+        input_amount: u64,
+        is_sell: bool,
+    ) acquires RewardsVaultSignerCapability {
+
+        // Simulate swap to get quote volume, then execute swap.
+        let swapper_address = signer::address_of(swapper);
+        let swap = emojicoin_dot_fun::simulate_swap(
+            swapper_address,
+            market_address,
+            input_amount,
+            is_sell,
+            @integrator,
+            INTEGRATOR_FEE_RATE_BPS,
+        );
+        let ( _, _, _, _, _, _, _, _, _, _, quote_volume, _, _, _, _, _) =
+            emojicoin_dot_fun::unpack_swap(swap);
+        emojicoin_dot_fun::swap<Emojicoin, EmojicoinLP>(
+            swapper,
+            market_address,
+            input_amount,
+            is_sell,
+            @integrator,
+            INTEGRATOR_FEE_RATE_BPS,
+        );
+
+        // Return if quote volume is below threshold, otherwise proceed to lottery.
+        if (quote_volume < VOLUME_THRESHOLD_IN_OCTAS) return;
+
+        // Get vault balance, returning without lottery if vault is empty.
+        let vault_signer_cap_ref =
+            &borrow_global<RewardsVaultSignerCapability>(@emojicoin_dot_fun).signer_capability;
```

Comment:
```suggestion
            &borrow_global<RewardsVaultSignerCapability>(@rewards).signer_capability;
```

Getting this error message right now:

```shell
{
  "Error": "Simulation failed with status: Execution failed in 0x76044a237dcc3f71af75fb314f016e8032633587f7d70df4e70777f2b0221e75::emojicoin_dot_fun_rewards::swap_with_rewards at code offset 42\nExecution failed with message: Failed to borrow global resource from babe8f471b7f4744502b1397530bafe80e3731b358c0dfeba26b38b2603bd00d"
}
```

Pretty sure it's because of this line, looks like ya used the old contract ;)

Reply:
Addressed in 7c3c6fb